### ISSUE TITLE
Standard predictor is predicting from Jan 11, 2021

### ISSUE THREAD

**Issue by ofrancon** [leaf-ai/covid-xprize#186]
Standard predictor is predicting the exact same number of cases, no matter what the start_date is.

How to reproduce:
Create 2 NPIs scenarios that always prescribe 0 for each NPI between start_date and end_date for one Country (e.g. Italy):

```
from covid_xprize.scoring.predictor_scoring import load_dataset
from covid_xprize.validation.scenario_generator import generate_scenario

LATEST_DATA_URL = 'https://raw.githubusercontent.com/OxCGRT/covid-policy-tracker/master/data/OxCGRT_latest.csv'
GEO_FILE = "countries_regions.csv"
latest_df = load_dataset(LATEST_DATA_URL, GEO_FILE)
```

July 2020 scenario:
```
# Apply zero NPIs between these dates
start_date_str = "2020-07-11"
end_date_str = "2020-07-31"
latest_df = load_dataset(LATEST_DATA_URL, GEO_FILE)
countries = ["Italy"]
scenario_df = generate_scenario(start_date_str, end_date_str, latest_df, countries, scenario="MIN")
# Truncate the scenario to end_date
end_date = pd.to_datetime(end_date_str, format='%Y-%m-%d')
scenario_df = scenario_df[scenario_df.Date <= end_date]
scenario_df.to_csv("prescriptions/20200711_20200731_italy_zero_ips.csv", index=False)
```

December 2020 scenario:
```
start_date_str = "2020-12-22"
end_date_str = "2021-01-11"
latest_df = load_dataset(LATEST_DATA_URL, GEO_FILE)
countries = ["Italy"]
scenario_df = generate_scenario(start_date_str, end_date_str, latest_df, countries, scenario="MIN")
# Truncate the scenario to end_date
end_date = pd.to_datetime(end_date_str, format='%Y-%m-%d')
scenario_df = scenario_df[scenario_df.Date <= end_date]
scenario_df.to_csv("prescriptions/20201222_20210111_italy_zero_ips.csv", index=False)

Predict for both of them using standard predictor:
```
from covid_xprize.standard_predictor.predict import predict

predict(start_date_str,
        end_date_str,
        "prescriptions/20200711_20200731_italy_zero_ips.csv",
        "prescriptions/20200711_20200731_italy_zero_preds.csv"
       )
```
and
```predict(start_date_str,
        end_date_str,
        "prescriptions/20201222_20210111_italy_zero_ips.csv",
        "prescriptions/20201222_20210111_italy_zero_preds.csv"
       )
```

Load the preds file and compare them: they have identical values in column `PredictedDailyNewCases`:
```
july_df = pd.read_csv("prescriptions/20200711_20200731_italy_zero_preds.csv",
                    parse_dates=['Date'],
                    dtype={"RegionName": str},
                    encoding="ISO-8859-1")
dec_df = pd.read_csv("prescriptions/20201222_20210111_italy_zero_preds.csv",
                    parse_dates=['Date'],
                    dtype={"RegionName": str},
                    encoding="ISO-8859-1")

july_df.compare(dec_df, keep_equal=True, keep_shape=True) 
```
Gives:
![image](https://user-images.githubusercontent.com/6790095/105083368-15f5f680-5a95-11eb-8340-d71d26790722.png)

**Comment by ofrancon** [comment]
Made standard predictor predict from the day before start_date. We had 2 issues:

The class level DataFrame containing the Oxford data was the one we trained on (Jan 11)
We were predicting from 2 days before start date instead of from the day before start date. For instance, if start date is July 11, we were predicting from the number of cases on July 9 instead of on July 10.
Fixes:
For 1., now truncating the Oxford DataFrame to start date, excluded. Also not storing this DataFrame at the class level anymore to avoid messing up potential future calls to predict(...).
Issue 2. was caused by the fact we don't know the outcome (y) for the last day when training, so we exclude it. But when predicting we should include it because we don't care about the y.
